GitUI/Objects/GitRemoteController.cs

```diff
@@ -144,18 +155,48 @@ public void LoadRemotes()
             {
                 Remotes.Clear();
 
-                var gitRemotes = _module.GetRemotes().Where(x => !string.IsNullOrWhiteSpace(x)).ToList();
-                if (gitRemotes.Any())
+                Action<List<GitRemote>, bool> loadRemotes = (allRemotes, enabled) =>
                 {
-                    var remotes = gitRemotes.Select(remote => new GitRemote
+                    Func<string[]> func;
+                    if (enabled)
+                    {
+                        func = _module.GetRemotes;
+                    }
+                    else
+                    {
+                        func = _module.GetDisabledRemotes;
+                    }
+
+                    Func<string, string, bool, string> getSettingKey = (settingKey, remoteName, remoteEnabled) =>
                     {
-                        Name = remote,
-                        Url = _module.GetSetting(string.Format(SettingKeyString.RemoteUrl, remote)),
-                        Push = _module.GetSettings(string.Format(SettingKeyString.RemotePush, remote)).ToList(),
-                        PushUrl = _module.GetSetting(string.Format(SettingKeyString.RemotePushUrl, remote)),
-                        PuttySshKey = _module.GetSetting(string.Format(SettingKeyString.RemotePuttySshKey, remote)),
-                    }).ToList();
+                        var key = string.Format(settingKey, remoteName);
+                        return remoteEnabled ? key : ConfigFile.DisabledSectionPrefix + key;
+                    };
 
+                    var gitRemotes = func().Where(x => !string.IsNullOrWhiteSpace(x)).ToList();
+                    if (gitRemotes.Any())
+                    {
+                        allRemotes.AddRange(gitRemotes.Select(remote => new GitRemote
+                        {
+                            Disabled = !enabled,
+                            Name = remote,
+                            Url = _module.GetSetting(getSettingKey(SettingKeyString.RemoteUrl, remote, enabled)),
```

Comment:
Could you try to move the save/load logic to the GitRemote class.
static GitRemote.LoadFrom(ConfigFile configFile, remoteName, enabled)
static GitRemote.LoadFrom(ConfigSection remoteSection)
GitRemote.SaveTo(ConfigFile configFile)
GitRemote.SaveTo(ConfigSection remoteSection)